tests/gtest_dlt_daemon_multiple_files_logging.cpp

```diff
@@ -0,0 +1,277 @@
+ * Mozilla Public License (MPL), v. 2.0.
+ * If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * For further information see http://www.genivi.org/.
+ */
+
+/*!
+ * \author
+ * Oleg Tropmann <oleg.tropmann@daimler.com>
+ * Daniel Weber <daniel.w.weber@daimler.com>
+ *
+ * \copyright Copyright © 2022 Daimler TSS GmbH. \n
+ * License MPL-2.0: Mozilla Public License version 2.0 http://mozilla.org/MPL/2.0/.
+ *
+ * \file gtest_dlt_daemon_multiple_files_logging.ccpp
+ */
+
+#include <gtest/gtest.h>
+
+int connectServer(void);
+
+extern "C"
+{
+#include "dlt_common.h"
+#include <syslog.h>
+#include <dirent.h>
+#include <string.h>
+}
+
+// publish prototypes
+void configure(const char* path, const char* file_name, bool enable_limit, int file_size, int max_files_size);
+void write_log_message();
+void verify_multiple_files(const char* path, const char* file_name, int file_size, int max_files_size);
+void verify_single_file(const char* path, const char* file_name);
+void verify_in_one_file(const char* path, const char* file_name, const char* log1, const char* log2);
+bool file_contains_strings(const char* abs_file_path, const char* str1, const char* str2);
+int get_file_index(char* file_name);
+int compare_int(const void* a, const void* b);
+
+/**
+ * Configure dlt logging using file size limits.
+ */
+TEST(t_dlt_logging_multiple_files, normal)
+{
+    const char* path = "/tmp";
+    const char* file_name = "dlt.log";
+    const int file_size = 128;
+    const int max_file_size = 512;
+    configure(path, file_name, true, file_size, max_file_size);
+    write_log_message();
+    EXPECT_NO_THROW(dlt_log_free());
+    verify_multiple_files(path, file_name, file_size, max_file_size);
+}
+
+/**
+ * Configure dlt logging using file size limits.
+ * Though, due to an error during initialization dlt logging defaults to one file logging.
+ */
+TEST(t_dlt_logging_one_file_as_fallback, normal)
+{
+    const char* path = "/tmp";
+    const char* file_name = "dltlog";
+    configure(path, file_name, true, 128, 512);
+    write_log_message();
+    EXPECT_NO_THROW(dlt_log_free());
+    verify_single_file(path, file_name);
+}
+
+/**
+ * Configure dlt logging without file size limits resulting in one file logging.
+ */
+TEST(t_dlt_logging_one_file, normal)
+{
+    const char* path = "/tmp";
+    const char* file_name = "dlt.log";
+    configure(path, file_name, false, 128, 512);
+    write_log_message();
+    EXPECT_NO_THROW(dlt_log_free());
+    verify_single_file(path, file_name);
+}
+
+/**
+ * The dlt_daemon calls dlt_log_init multiple times. In the past, so we create
+ * unnecessary two files. The reinit have to append to the first file.
+ */
+TEST(t_dlt_logging_multiple_files_append_reinit, normal)
+{
+    const char* path = "/tmp";
+    const char* file_name = "dlt.log";
+    const int file_size = 256;
+    const int max_file_size = 512;
+
+    const char* log1 = "ONE\n";
```

Comment:
Hello @danielweber2018 
Build logs show me:
```
Scanning dependencies of target gtest_dlt_daemon_multiple_files_logging
[ 86%] Building CXX object tests/CMakeFiles/gtest_dlt_daemon_multiple_files_logging.dir/gtest_dlt_daemon_multiple_files_logging.cpp.o
/home/lum3hc/work/github/PR442/dlt-daemon/tests/gtest_dlt_daemon_multiple_files_logging.cpp: In member function ‘virtual void t_dlt_logging_multiple_files_append_reinit_normal_Test::TestBody()’:
/home/lum3hc/work/github/PR442/dlt-daemon/tests/gtest_dlt_daemon_multiple_files_logging.cpp:106:28: warning: format not a string literal and no format arguments [-Wformat-security]
     dlt_vlog(LOG_INFO, log1);
                            ^
/home/lum3hc/work/github/PR442/dlt-daemon/tests/gtest_dlt_daemon_multiple_files_logging.cpp:110:28: warning: format not a string literal and no format arguments [-Wformat-security]
     dlt_vlog(LOG_INFO, log2);
                            ^
[ 87%] Linking CXX executable gtest_dlt_daemon_multiple_files_logging
```
Could you please consider to use string literal format?
Maybe char array?